.github/workflows/todoapp-ci-pipeline.yml

```
@@ -16,6 +16,32 @@ jobs:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v4
-    - run: docker build --target test --tag todo-app:test .
-    - run: docker run todo-app:test
\ No newline at end of file
+      - uses: actions/checkout@v4
+      - run: docker build --target test --tag todo-app:test .
+      - run: docker run todo-app:test
+
+  push_to_dockerhub:
+    name: Push docker image and deploy
+    runs-on: ubuntu-latest
+    needs: build
+    if: github.ref == 'refs/heads/exercise-9-expanding-ci-pipeline'
+    steps:
+      - uses: actions/checkout@v4
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build Docker image
+        run: |
+         docker build --target production --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest \
+                      --target production --tag ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:${{ github.sha }} .
+
+      - name: Push Docker image
+        run: | 
+          docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest
+          docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:${{ github.sha }}

```

Not sure it's necessarily better, but in case it's useful info, I believe this could also be done like:
```suggestion
          docker push ${{ secrets.DOCKERHUB_USERNAME }}/todo-app:latest --all-tags
```